api_app/analyzers_manager/file_analyzers/yara_scan.py

```diff
@@ -3,60 +3,49 @@
 
 import io
 import logging
-import os
 import zipfile
-from typing import List, Tuple
+from pathlib import PosixPath
+from typing import List, Optional, Tuple
+from urllib.parse import urlparse
 
 import requests
 import yara
+from cache_memoize import cache_memoize
+from django.conf import settings
 from git import Repo
 
 from api_app.analyzers_manager.classes import FileAnalyzer
 from api_app.analyzers_manager.dataclasses import AnalyzerConfig
 from api_app.exceptions import AnalyzerRunException
+from intel_owl.settings._util import set_permissions
 
 logger = logging.getLogger(__name__)
 
 
 class YaraScan(FileAnalyzer):
     def set_params(self, params):
-        self.directories_with_rules = params.get("directories_with_rules", [])
-        self.recursive = params.get("recursive", False)
         self.result = []
-        self.ruleset: List[Tuple[str, yara.Rules]] = []
-
-    def load_directory(self, rulepath):
-        # if you do not have an index file,...
-        # .. just extract all the rules in the .yar files
-        for f in os.listdir(rulepath):
-            full_path = f"{rulepath}/{f}"
-            if os.path.isfile(full_path):
-                try:
-                    if (
-                        full_path.endswith(".yar")
-                        or full_path.endswith(".yara")
-                        or full_path.endswith(".rule")
-                    ):
-                        self.ruleset.append(
-                            (
-                                full_path,
-                                yara.compile(
-                                    full_path,
-                                    externals={"filename": self.filename},
-                                ),
-                            )
-                        )
-                    elif full_path.endswith(".yas"):
-                        self.ruleset.append((full_path, yara.load(full_path)))
-                except yara.SyntaxError as e:
-                    logger.warning(f"Rule {full_path} " f"has a syntax error {e}")
-                    continue
+        self.ignore_rules = params.get("ignore", [])
```

Comment:
to allow the same previous possible usage we should also add something like `self.selected_repos` that as a default contains all the available repos (so all the yara rules are used) but it is customizable at runtime and people could de-select the ones that they do not want to include so they can just execute the ones that they want

(this could make sense for performance reasons. Plus some yara rules are meant to be used in completely different contexts of other rules.)